lib/spack/spack/version.py

```diff
@@ -165,15 +169,27 @@ def __gt__(self, other):
         return not self.__lt__(other)
 
 
-class Version(object):
+def is_git_version(string):
+    if string.startswith('git.'):
+        return True
+    elif len(string) == 40 and COMMIT_VERSION.match(string):
+        return True
+    return False
```

Comment:
Doesn't it make sense to also allow `git.<commit hash>` for consistency?  I can see people trying to make commits that way based on branches and failing mysteriously.